DAOS-3906 Add DAOS DFS Java API and implement Hadoop FileSystem Interface

zjf2012 opened the pull request:
This PR is to continue the Java Wrapper. It includes,

- upgraded log4j to log4j2 to fix log4j vulnerability.
- fixed JVM crash when shutdown
- fixed miscellaneous issues from review. They are,

1, remove pool creation related code
2, add the move function with parameter of opened source dir and target dir
3, use object type name instead of id
4, more comments in Constants and StatAttributes
5, newline after declaration

zjf2012 commented:
@johannlombardi @mchaarawi,  I addressed some issues listed in the PR description. please review.

thanks

daosbuild1 commented on src/client/java/daos-java-api/src/main/native/com_intel_daos_client_DaosFsClient.c:
```diff
@@ -606,6 +536,50 @@ Java_com_intel_daos_client_DaosFsClient_move(JNIEnv *env, jobject obj,
+ * \param[in]	srcName		source name, just file name without any path
+ * \param[in]	destPrtObjId	dest parent object id
+ * \param[in]	destName	destination name, just file name without any
+ * 				path
```
(style)  please, no space before tabs

daosbuild1 replied:
(style)  please, no space before tabs

daosbuild1 replied:
(style)  please, no space before tabs

daosbuild1 commented on src/client/java/daos-java-api/src/main/native/com_intel_daos_client_DaosFsClient.c:
```diff
@@ -606,6 +536,50 @@ Java_com_intel_daos_client_DaosFsClient_move(JNIEnv *env, jobject obj,
+ * 				path
+ */
+JNIEXPORT void JNICALL
+Java_com_intel_daos_client_DaosFsClient_move__JJLjava_lang_String_2JLjava_lang_String_2(
```
(style)  line over 80 characters

daosbuild1 replied:
(style)  line over 80 characters

daosbuild1 replied:
(style)  line over 80 characters

daosbuild1 commented on src/client/java/daos-java-api/src/main/native/com_intel_daos_client_DaosFsClient.c:
```diff
@@ -606,6 +536,50 @@ Java_com_intel_daos_client_DaosFsClient_move(JNIEnv *env, jobject obj,
+{
+	char *src_base = (*env)->GetStringUTFChars(env, srcName, NULL);
+	char *dest_base = (*env)->GetStringUTFChars(env, destName, NULL);
+	dfs_obj_t *src_dir_handle = *(dfs_obj_t**)&srcPrtObjId;
```
(style)  "(foo**)" should be "(foo **)"

daosbuild1 commented on src/client/java/daos-java-api/src/main/native/com_intel_daos_client_DaosFsClient.c:
```diff
@@ -606,6 +536,50 @@ Java_com_intel_daos_client_DaosFsClient_move(JNIEnv *env, jobject obj,
+	char *src_base = (*env)->GetStringUTFChars(env, srcName, NULL);
+	char *dest_base = (*env)->GetStringUTFChars(env, destName, NULL);
+	dfs_obj_t *src_dir_handle = *(dfs_obj_t**)&srcPrtObjId;
+	dfs_obj_t *dest_dir_handle = *(dfs_obj_t**)&destPrtObjId;
```
(style)  "(foo**)" should be "(foo **)"

daosbuild1 commented on src/client/java/daos-java-api/src/main/native/com_intel_daos_client_DaosFsClient.c:
```diff
@@ -540,33 +465,36 @@ Java_com_intel_daos_client_DaosFsClient_daosFinalize(JNIEnv *env,
 JNIEXPORT void JNICALL
-Java_com_intel_daos_client_DaosFsClient_move(JNIEnv *env, jobject obj,
-		jlong dfsPtr, jstring srcPath, jstring destPath)
+Java_com_intel_daos_client_DaosFsClient_move__JLjava_lang_String_2Ljava_lang_String_2(
```
(style)  line over 80 characters

daosbuild1 replied:
(style)  line over 80 characters

daosbuild1 replied:
(style)  line over 80 characters

daosbuild1 commented on src/client/java/daos-java-api/src/main/native/com_intel_daos_client_DaosFsClient.c:
```diff
@@ -606,6 +536,50 @@ Java_com_intel_daos_client_DaosFsClient_move(JNIEnv *env, jobject obj,
+			dest_base, NULL);
+
+	if (rc) {
+		char *tmp = "Failed to move source file (%s) under dir (%ld)" \
```
(style)  Avoid unnecessary line continuations

daosbuild1 commented on src/client/java/daos-java-api/src/main/native/com_intel_daos_client_DaosFsClient.c:
```diff
@@ -606,6 +536,50 @@ Java_com_intel_daos_client_DaosFsClient_move(JNIEnv *env, jobject obj,
+	dfs_obj_t *src_dir_handle = *(dfs_obj_t**)&srcPrtObjId;
+	dfs_obj_t *dest_dir_handle = *(dfs_obj_t**)&destPrtObjId;
+	mode_t tmp_mode;
+	dfs_t *dfs = *(dfs_t**)&dfsPtr;
```
(style)  "(foo**)" should be "(foo **)"

daosbuild1 commented on src/client/java/daos-java-api/src/main/native/include/com_intel_daos_client_DaosFsClient.h:
```diff
@@ -26,10 +34,10 @@ JNIEXPORT void JNICALL Java_com_intel_daos_client_DaosFsClient_mkdir
  */
 JNIEXPORT jlong JNICALL Java_com_intel_daos_client_DaosFsClient_createNewFile
-  (JNIEnv *, jobject, jlong, jstring, jstring, jint, jint, jint, jint, jboolean);
+  (JNIEnv *, jobject, jlong, jstring, jstring, jint, jint, jstring, jint, jboolean);
```
(style)  line over 80 characters

daosbuild1 replied:
(style)  line over 80 characters

daosbuild1 replied:
(style)  line over 80 characters

daosbuild1 commented on src/client/java/daos-java-api/src/main/native/include/com_intel_daos_client_DaosFsClient.h:
```diff
@@ -12,9 +12,17 @@ extern "C" {
+ * Method:    move
+ * Signature: (JJLjava/lang/String;JLjava/lang/String;)V
+ */
+JNIEXPORT void JNICALL Java_com_intel_daos_client_DaosFsClient_move__JJLjava_lang_String_2JLjava_lang_String_2
```
(style)  line over 80 characters

daosbuild1 replied:
(style)  line over 80 characters

daosbuild1 replied:
(style)  line over 80 characters

daosbuild1 commented on src/client/java/daos-java-api/src/main/native/include/com_intel_daos_client_DaosFsClient.h:
```diff
@@ -12,9 +12,17 @@ extern "C" {
  * Signature: (JLjava/lang/String;Ljava/lang/String;)V
  */
-JNIEXPORT void JNICALL Java_com_intel_daos_client_DaosFsClient_move
+JNIEXPORT void JNICALL Java_com_intel_daos_client_DaosFsClient_move__JLjava_lang_String_2Ljava_lang_String_2
```
(style)  line over 80 characters

daosbuild1 replied:
(style)  line over 80 characters

daosbuild1 replied:
(style)  line over 80 characters

daosbuild1 commented on src/client/java/daos-java-api/find_java_home.sh:
```diff
@@ -1,17 +1,19 @@
-	java_path=$(dirname $(readlink -f $(which java)))
-	java_home=$(echo $java_path | sed 's^/jre/bin^^')
-	if  [ ! -f "${java_home}/include/jni.h" ] 
+	java_path="$(dirname $(readlink -f $(which java)))"
```
(lint) Quote this to prevent word splitting. [SC2046]
(lint) Quote this to prevent word splitting. [SC2046]

daosbuild1 replied:
(lint) Quote this to prevent word splitting. [SC2046]
(lint) Quote this to prevent word splitting. [SC2046]

daosbuild1 replied:
(lint) Quote this to prevent word splitting. [SC2046]
(lint) Quote this to prevent word splitting. [SC2046]

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-1746/1/execution/node/47/log

daosbuild1 commented on src/client/java/daos-java-api/src/main/native/com_intel_daos_client_DaosFsClient.c:
```diff
@@ -606,6 +536,50 @@ Java_com_intel_daos_client_DaosFsClient_move(JNIEnv *env, jobject obj,
+			dest_base, NULL);
+
+	if (rc) {
+		char *tmp = "Failed to move source file (%s) under dir (%ld)" \
```
(style)  Avoid unnecessary line continuations

daosbuild1 replied:
(style)  Avoid unnecessary line continuations

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-1746/2/execution/node/48/log

zjf2012 commented:
> Could you please give me an updated commit message summarizing all the incremental changes you have made in this PR? Thanks in advance.

I listed them in the PR description field. I copied them here.

This PR is to continue the Java Wrapper. It includes,

- upgraded log4j to log4j2 to fix log4j vulnerability.
- fixed JVM crash when shutdown
- fixed miscellaneous issues from review. They are,
1, remove pool creation related code
2, add the move function with parameter of opened source dir and target dir
3, use object type name instead of id
4, more comments in Constants and StatAttributes
5, newline after declaration

johannlombardi commented:
oh, nm. I missed it.

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-1746/3/execution/node/48/log
